floor_generator/lib/misc/extension/foreign_key_action.dart

```diff
@@ -18,9 +15,8 @@ abstract class ForeignKeyAction {
         return 'SET DEFAULT';
       case ForeignKeyAction.cascade:
         return 'CASCADE';
-      case ForeignKeyAction.noAction:
       default:
-        return 'NO ACTION';
+        return null;
```

Comment:
💡 We should change the `@nonNull` annotation to `@nullable` if returning `null` is really what we want here. Have you thought of throwing an `Error` instead? I'm thinking about this approach because we're in full control of the cases (`ForeignKeyAction` enum) so we *should* never see the error in "production".

Some background about the `@nonNull` and `@nullable` annotations: At some point back in time, I heard that the Dart team at Google is working on making the language null-safe. The time is about to arrive as [non-nullable types are currently in preview](https://medium.com/dartlang/announcing-sound-null-safety-defd2216a6f3) and will be released later this year. In order for us to have an easy migration, the annotations will help 🙂

Reply:
This code is now throwing an ArgumentError.